cdap-hbase-compat-base/src/main/java/co/cask/cdap/data2/util/hbase/DefaultHBaseDDLExecutor.java

```diff
@@ -205,4 +210,49 @@ public void close() throws IOException {
       admin.close();
     }
   }
+
+  protected abstract void doGrantPermissions(String namespace, String name,
+                                             Map<String, Permission.Action[]> permissions) throws IOException;
+
+  @Override
+  public void grantPermissions(String namespace, String name, Map<String, String> permissions) throws IOException {
+    Map<String, Permission.Action[]> privilegesToGrant = new HashMap<>(permissions.size());
+    for (Map.Entry<String, String> entry : permissions.entrySet()) {
+      String user = entry.getKey();
+      String actionsForUser = entry.getValue();
+      try {
+        privilegesToGrant.put(user, toActions(actionsForUser));
+      } catch (IllegalArgumentException e) {
+        throw new IOException(String.format("Error granting permissions '%s' for table %s:%s to user %s: %s",
+                                            actionsForUser, namespace, name, user, e.getMessage()));
+      }
+      doGrantPermissions(namespace, name, privilegesToGrant);
+    }
+  }
+
+  protected Permission.Action[] toActions(String permissions) {
+    Permission.Action[] actions = new Permission.Action[permissions.length()];
+    for (int i = 0; i < actions.length; i++) {
+      actions[i] = toAction(permissions.charAt(i));
+    }
+    return actions;
+  }
+
+  @Nullable
```

Comment:
good catch. I first had it return null so that the caller can throw a detailed exception. Then changed it throw here, and forgot the @Nullable.